### PR TITLE
Cleanup client-count-utils tests to prep for refactor

### PR DESCRIPTION
**Pull request by hellobontempo**
In preparation for some client count util refactoring I used mock data to generate a response, then passed it to each helper function to capture the expected shape. 

I was the one who originally wrote these util tests, and in hindsight they weren't really testing for things that I wanted them to.

### CODE REVIEW

File: ui/tests/integration/utils/client-count-utils-test.js [hunk 1]
```diff
@@ -7,595 +7,285 @@ import { module, test } from 'qunit';
 import { setupTest } from 'ember-qunit';
 import {
   filterVersionHistory,
-  flattenDataset,
```
Comment (hellobontempo) [inline ui/tests/integration/utils/client-count-utils-test.js:10]: In a follow on PR I've refactored to no longer need this helper, deleting the test here

File: ui/tests/integration/utils/client-count-utils-test.js [hunk 2]
```diff
@@ -7,595 +7,285 @@ import { module, test } from 'qunit';
...
-              clients: 5,
-            },
-          },
-        ],
-      },
-    ],
-    new_clients: {
+import { addMonths } from 'date-fns';
+
+const RESPONSE = {
```
Comment (hellobontempo) [inline ui/tests/integration/utils/client-count-utils-test.js:19]: I assigned the response as a const here for reference, but also so we can confirm the numbers don't change. (Mirage generates new numbers each query). It's looooong but seems reasonable to have in this test as it helps discern how the response shape is changed to work for charts, etc.

Comment (hashishaw) [reply ui/tests/integration/utils/client-count-utils-test.js:19]: My only thought is to move this to a separate helper file (like we do with pki/values) so it doesn't take up so much room. But +1 to having consistent values!

**Review by hashishaw — APPROVED** [review]
😎 just one suggestion, otherwise LGTM

File: ui/tests/integration/utils/client-count-utils-test.js (hunk 2 above)
Comment (hellobontempo) [reply ui/tests/integration/utils/client-count-utils-test.js:19]: I agree! I'll do that in the next PR!

**Pull request merged**